contracts/common/SafeERC20.sol

```diff
@@ -75,7 +75,16 @@ library SafeERC20 {
             )
 
             if gt(success, 0) {
-                ret := mload(ptr)
+                switch returndatasize
+
+                // 32 bytes returned; is valid return
+                case 0x20 {
+                    ret := mload(ptr)
+                }
+                // Else, mark call as failed
+                default {
+                    success := 0
```

Comment:
Nitpick: The revert reasons in `staticBalanceOf` and `staticAllowance` denote that the call to the token reverted, which is no longer always the case with this change. I don't think we should change the revert reason but we should at least make a note for this.

Reply:
The comment in the function header says "Reverts if the call fails for some reason (should never fail)", we could just update it here.

Reply:
Modified to "(should never fail for correctly implemented ERC20s)"